contracts/modules/cover/Cover.sol

```diff
@@ -781,17 +800,22 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return uint(activeCover[assetId].totalActiveCoverInAsset);
   }
 
-  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
+  // Returns true if the product exists and the pool is authorized to have the product
+  function isPoolAllowed(uint productId, uint poolId) public view returns (bool) {
 
     uint poolCount = allowedPools[productId].length;
 
+    // If no pools are specified, nothing is blacklisted
     if (poolCount == 0) {
-      return true;
+      // If the product is deprecated, it's not allowed
+      // If the product doesn't exist, it's not allowed
+      return  productId < _products.length && !_products[productId].isDeprecated;
```

Comment:
still thinking this can be a separate predicate

It can be an internal function so it gets inlined